vulture_os/toolkit/api_parser/apex/apex.py

```diff
@@ -0,0 +1,177 @@
+               'iat': iat,
+               'version': version,
+               'checksum': self.create_checksum(http_method, raw_url, headers, request_body)}
+        token = jwt.encode(payload, api_key, algorithm=algorithm)
+        return token
+
+    def _connect(self, raw_url):
+        try:
+            # New session because timeout is very short
+            self.session = requests.Session()
+            jwt_token = self.create_jwt_token(self.apex_application_id, self.apex_api_key, 'GET',
+                              raw_url,
+                              self.canonicalRequestHeaders, self.useRequestBody, iat=timezone.now().timestamp())
+            self.session.headers.update(self.HEADERS)
+            self.session.headers.update({'Authorization': 'Bearer ' + jwt_token })
+
+            return True
+
+        except Exception as err:
+            raise ApexAPIError(err)
+
+    def test(self):
+        start_time = timezone.now() - timedelta(days=1)
+        try:
+            status, logs, next_time = self.get_logs("data_loss_prevention", start_time)
+
+            if not status:
+                return {
+                    "status": False,
+                    "error": logs
+                }
+
+            return {
+                "status": True,
+                "data": logs
+            }
+        except Exception as e:
+            logger.exception(f"[{__parser__}]:test: {e}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+        
+    def execute_query(self, url, timeout=60):
+
+        response = self.session.get(
+            url,
+            proxies=self.proxies,
+            timeout=timeout,
+            verify=self.api_parser_custom_certificate if self.api_parser_custom_certificate else self.api_parser_verify_ssl
+        )
+        if response.status_code != 200:
+            error = f"Error at Apex API Call: status : {response.status_code}, content : {response.content}"
+            logger.error(f"[{__parser__}]:get_logs: {error}", extra={'frontend': str(self.frontend)})
+            raise ApexAPIError(error)
+        
+        return response.json()
+
+
+    def get_logs(self, kind, since):
+
+        url_path = f"/WebApp/api/v1/Logs/{kind}?output_format=CEF&since_time={int(since.timestamp())}"
+        logs = []
+        try:
+            while True:
+                self._connect(url_path)
+                content = self.execute_query(self.apex_server_host + url_path)
+                if data_logs := content["Data"]["Logs"]:
+                    logs.extend(data_logs)
+                if next_url := content["Data"].get("Next"):
+                    url_path = f"/WebApp/api/v1/Logs/{kind}" + next_url[next_url.index("?"):]
+                else:
+                    break
+            next_timestamp = int(content["Data"]["NextPage"]["SinceTime"])
+
+        except Exception as e:
+            return False, e, None
```

Comment:
The exception as 'e' is never used. You catched the error but  you never pass it to the "raise ApexAPIError" line 163.